src/go/rdctl/pkg/shutdown/shutdown.go

```diff
@@ -54,72 +60,51 @@ func FinishShutdown(waitForShutdown bool) error {
 	return nil
 }
 
-func doCheckWithTimeout(checkFunc func() bool, killFunc func() error, waitForShutdown bool, retryCount int, retryWait int, operation string) error {
-	for iter := 0; waitForShutdown && iter < retryCount; iter++ {
+func (s *ShutdownInfo) waitForAppToDieOrKillIt(checkFunc func() (bool, error), killFunc func() error, retryCount int, retryWait int, operation string) error {
+	for iter := 0; s.waitForShutdown && iter < retryCount; iter++ {
 		if iter > 0 {
 			logrus.Debugf("checking %s showed it's still running; sleeping %d seconds\n", operation, retryWait)
 			time.Sleep(time.Duration(retryWait) * time.Second)
 		}
-		if !checkFunc() {
+		status, err := checkFunc()
+		if err != nil {
+			return fmt.Errorf("checking operation %s => error %w", operation, err)
```

Comment:
Sample output:
> checking operation the app => error The file was not found.

The wording seems a bit odd; consider `while checking %s, found error: %w`.